HLSPlugin/src/com/kaltura/hls/m2ts/PESProcessor.as

```diff
@@ -131,8 +137,17 @@ package com.kaltura.hls.m2ts
                     types[pid] = type;
                     seenPIDsByClass[mediaClass] = pid;
                 }
-                
+                //check if we see bytes[cursor]..infolength find ID3 string - if we find we need to save the PID
+				//PES header - find the packet size
+				//
                 // Skip the esInfo data.
+				oldPosition = bytes.position;
+				var ID3Index:Number = indexOf(bytes,"ID3",cursor);
+				if (ID3Index > 0){
+					lastID3Point = pid;
```

Comment:
If I understand this logic, you generally scan all ES info data for ID3 and if you see it, note the PID the "lastID3Point"?